pkg/controller/replication/replication_controller_test.go

```diff
@@ -889,3 +889,50 @@ func TestOverlappingRCs(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkGetPodControllers(b *testing.B) {
+	client := client.NewOrDie(&client.Config{Host: "", GroupVersion: testapi.Default.GroupVersion()})
+	manager := NewReplicationManager(client, controller.NoResyncPeriodFunc, BurstReplicas)
+
+	pods := []api.Pod{}
+	for i := 0; i < 100; i++ {
```

Comment:
Can you make the magic numbers into constants?

Also, I suggest trying e.g. 1000 namespaces each with 10 RCs each with 10 pods.